src/Http/Event/ImportEventRequestHandler.php

```diff
@@ -52,37 +99,173 @@ public function handle(ServerRequestInterface $request): ResponseInterface
+            $workflowStatus = WorkflowStatus::READY_FOR_VALIDATION();
+        }
+
+        $commands = [];
+        if (!$eventExists) {
+            $eventAggregate = EventAggregate::create(
+                $eventId,
+                $mainLanguage,
+                $title,
+                $type,
+                $location,
+                $calendar,
+                $theme,
+                $publishDate
+            );
+
+            if ($workflowStatus->sameAs(WorkflowStatus::READY_FOR_VALIDATION())) {
+                $eventAggregate->publish($publishDate);
+            }
+
+            // Events created by specific API partners should automatically be approved.
+            $consumer = $this->getConsumer($request);
+            if ($consumer && $this->shouldApprove->satisfiedBy($consumer)) {
+                if (!str_contains($request->getUri()->getPath(), 'imports')) {
+                    $eventAggregate->publish($publishDate);
+                }
+
+                $eventAggregate->approve();
+            }
+
+            $this->aggregateRepository->save($eventAggregate);
+        } else {
+            if ($workflowStatus->sameAs(WorkflowStatus::READY_FOR_VALIDATION())) {
+                $commands[] = new Publish($eventId, $publishDate);
+            }
 
-        try {
-            $commandId = $this->documentImporter->import($document);
-        } catch (DBALEventStoreException $exception) {
-            if ($exception->getPrevious() instanceof UniqueConstraintViolationException) {
-                throw ApiProblem::resourceIdAlreadyInUse($eventId);
+            $commands[] = new UpdateTitle($eventId, $mainLanguage, $title);
+            $commands[] = new UpdateType($eventId, $type->getId());
+            $commands[] = new UpdateLocation($eventId, $location);
+            $commands[] = new UpdateCalendar($eventId, $calendar);
+
+            if ($theme) {
+                $commands[] = new UpdateTheme($eventId, $theme->getId());
+            }
+        }
+
+        if ($location->isDummyPlaceForEducation()) {
+            $audienceType = AudienceType::education();
+        } else {
+            $audienceType = $event->getAudienceType();
+        }
+        $commands[] = new UpdateAudience($eventId, $audienceType);
+
+        $bookingInfo = $eventAdapter->getBookingInfo();
+        $commands[] = new UpdateBookingInfo($eventId, $bookingInfo);
+
+        $contactPoint = $eventAdapter->getContactPoint();
+        $commands[] = new UpdateContactPoint($eventId, $contactPoint);
+
+        $description = $eventAdapter->getDescription();
+        if ($description) {
+            $commands[] = new UpdateDescription($eventId, $mainLanguage, $description);
+        }
+
+        // Update the organizer only at the end, because it can trigger UiTPAS to send messages to another worker
+        // which might cause race conditions if we're still dispatching other commands here as well.
```

Comment:
This comment is a bit contradictory because it says that the Organizer should only be updated at the very end, but there are still quite some commands after it.

Reply:
In the old `EventDocumentImporter` it was also at the end.

Reply:
Oh nice catch, I must have done something wrong when porting the code. I will put this at the end for both the event and place imports.